docs/src/main/asciidoc/hibernate-orm-panache-kotlin.adoc

```diff
@@ -0,0 +1,204 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Simplified Hibernate ORM with Panache and Kotlin
+
+include::./attributes.adoc[]
+:config-file: application.properties
+
+Hibernate ORM is the de facto standard JPA implementation and is well-known in the Java ecosystem.  Panache offers a
+new layer atop this familiar framework.  This guide will not dive in to the specifics of either as those are already
+covered in the link:hibernate-orm-panache.adoc[Panache guide].  In this guide, we'll cover the Kotlin specific changes
+needed to use Panache in your Kotlin-based Quarkus applications.
+
+== First: an example
+
+As we saw in the Panache guide, Panache allows us to extend the functionality in our entities and DAOs with some automatically
+provided functionality.  When using Kotlin, the approach is very similar to what we see in the Java version with a slight
+change or two.  To Panache-enable your entity, you would define it something like:
+
+[source,kotlin]
+----
+@Entity
+class Person: PanacheEntity {
+    lateinit var name: String
+    lateinit var birth: LocalDate
+    lateinit var status: Status
+}
+----
+
+As you can see our entities remain simple there is, however, a slight difference from the Java version.  The Kotlin
+language doesn't support the notion of static methods in quite the same way as Java does.  Instead, we use the
+[companion object](https://kotlinlang.org/docs/tutorials/kotlin-for-py/objects-and-companion-objects.html#companion-objects):
+
+[source,kotlin]
+----
+@Entity
+class Person : PanacheEntity {
+    companion object: PanacheCompanion<Person> {  // <1>
+        fun findByName(name: String) = find("name", name).firstResult()
+        fun findAlive() = list("status", Status.Alive)
+        fun deleteStefs() = delete("name", "Stef")
+    }
+
+    lateinit var name: String  // <2>
+    lateinit var birth: LocalDate
+    lateinit var status: Status
+}
+----
+<1> The companion object holds all the methods not related to a specific instance allowing for general management and
+querying bound to a specific type.
+<2> Here there are options, but we've chosen the `lateinit` approach.  This allows us to declare these fields as non-null
+knowing they will be properly assigned either by the constructor (not shown) or by hibernate loading data from the
+database.
+
+NOTE: Note that these types differ from the Java types mentioned in those tutorials.  For kotlin support, all the Panache
+types will be found in the `io.quarkus.hibernate.orm.panache.kotlin` package.  This subpackage allows for the distinction
+between the Java and Kotlin variants and allows for both to be used unambiguously in a single project.
+
+In the Kotlin version, we've simply moved the bulk of the link:https://www.martinfowler.com/eaaCatalog/activeRecord.html[`entity pattern`]
```

Comment:
```suggestion
In the Kotlin version, we've simply moved the bulk of the link:https://www.martinfowler.com/eaaCatalog/activeRecord.html[`active record pattern`]
```

For some reason, the Hibernate ORM with Panache guide seems to prefer this term, even though I don't agree with it.